File: recovered/src/com/lumiyaviewer/lumiya/render/spatial/DrawEntryList.java
package com.lumiyaviewer.lumiya.render.spatial;

import com.lumiyaviewer.lumiya.utils.InlineList;
import javax.annotation.Nullable;

/* loaded from: classes.dex */
public class DrawEntryList extends InlineList<DrawListEntry> {

    @Nullable
    private final EntryRemovalListener listener;

    public interface EntryRemovalListener {
        void onEntryRemovalRequested(DrawListEntry drawListEntry);
    }

    public DrawEntryList(@Nullable EntryRemovalListener entryRemovalListener) {
        this.listener = entryRemovalListener;
    }

    @Override // com.lumiyaviewer.lumiya.utils.InlineList
    public void requestEntryRemoval(DrawListEntry drawListEntry) {
        if (this.listener != null) {
            this.listener.onEntryRemovalRequested(drawListEntry);
        }
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/render/spatial/DrawEntryList.kt
package com.lumiyaviewer.lumiya.render.spatial

import com.lumiyaviewer.lumiya.utils.InlineList

class DrawEntryList(private val listener: EntryRemovalListener?) : InlineList<DrawListEntry>() {

    fun interface EntryRemovalListener {
        fun onEntryRemovalRequested(drawListEntry: DrawListEntry)
    }

    override fun requestEntryRemoval(drawListEntry: DrawListEntry) {
        listener?.onEntryRemovalRequested(drawListEntry)
    }
}
